server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenStoreTests.java

```diff
@@ -206,32 +210,46 @@ void retrieveNonExistentToken() {
 
     @Test
     void cleanUpExpiredTokensBasedOnExpiresField() {
+        TestClock clock = new TestClock();
+        Duration expirationTime = Duration.ofHours(1);
+        store = UaaTokenStore.constructorForTesting(dataSource, expirationTime, clock);
+
         int count = 10;
         String lastCode = null;
         for (int i = 0; i < count; i++) {
+            clock.instants.add(Instant.now());
             lastCode = store.createAuthorizationCode(clientAuthentication);
         }
         assertThat(jdbcTemplate.queryForObject("SELECT count(*) FROM oauth_code", Integer.class), is(count));
 
         jdbcTemplate.update("UPDATE oauth_code SET expiresat = ?", System.currentTimeMillis() - 60000);
 
         final String finalLastCode = lastCode;
+        clock.instants.add(Instant.now().plus(expirationTime));
+        clock.instants.add(Instant.now().plus(expirationTime));
+
         assertThrows(InvalidGrantException.class, () -> store.consumeAuthorizationCode(finalLastCode));
         assertThat(jdbcTemplate.queryForObject("SELECT count(*) FROM oauth_code", Integer.class), is(0));
     }
 
     @Test
     void cleanUpLegacyCodesCodesWithoutExpiresAtAfter3Days() {
+        TestClock clock = new TestClock();
+        store = UaaTokenStore.constructorForTesting(dataSource, Duration.ofMinutes(1), clock);
+
+
         int count = 10;
         long oneday = 1000 * 60 * 60 * 24;
         for (int i = 0; i < count; i++) {
             legacyCodeServices.createAuthorizationCode(clientAuthentication);
         }
         assertThat(jdbcTemplate.queryForObject("SELECT count(*) FROM oauth_code", Integer.class), is(count));
         jdbcTemplate.update("UPDATE oauth_code SET created = ?", new Timestamp(System.currentTimeMillis() - (2 * oneday)));
+        clock.instants.add(Instant.now());
         assertThrows(InvalidGrantException.class, () -> store.consumeAuthorizationCode("non-existent"));
```

Comment:
The three blocks of assertions make this test a bit hard to follow. Maybe it wants to be two or three different tests?

Reply:
Same as the other comment about code I didn't change.

Reply:
You did make the code blocks longer, though. An easy fix might be adding some whitespace.